cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -257,43 +265,91 @@ public Catalog getLocationCatalog(String cursor, int pageSize, String unitSystem
     }
 
     @Override
-    public Catalog getLocationCatalog(String cursor, int pageSize, String unitSystem, String office,
+    public Catalog getLocationCatalog(String page, int pageSize, String unitSystem, String office,
                                       String idLike, String categoryLike, String groupLike) {
         //Now querying against AV_LOC2 as it gives us back the same information as querying against
         //location group views. This makes the code clearer and improves performance.
         //If there is a performance improvement by switching back to location groups and querying against
         //location codes (previous implementation used location_id) for joins, feel free to implement.
+        final AV_LOC2 avLoc2 = AV_LOC2.AV_LOC2;
         String locCursor = "*";
-        Condition condition = buildCatalogWhere(unitSystem, office, idLike, categoryLike, groupLike);
+        String searchOffice = office;
+        String curOffice = null;
+        Catalog.CatalogPage catPage = null;
         int total;
-        if (cursor == null || cursor.isEmpty()) {
+        Condition condition = avLoc2.LOCATION_ID.upper().likeRegex(idLike.toUpperCase())
```

Comment:
Previously `idLike` was null checked.

Reply:
that overload is still a problem though...

Reply:
oy, now I see it.